nexus-repository-helm-it/src/test/java/org/sonatype/nexus/plugins/helm/internal/cleanup/CleanupTaskHelmIT.java

```diff
@@ -0,0 +1,119 @@
+import org.ops4j.pax.exam.Option;
+
+import static java.lang.String.format;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.sonatype.nexus.plugins.helm.internal.HelmITSupport.MONGO_PKG_FILE_NAME_404_TGZ;
+import static org.sonatype.nexus.plugins.helm.internal.HelmITSupport.MONGO_PKG_FILE_NAME_600_TGZ;
+import static org.sonatype.nexus.plugins.helm.internal.HelmITSupport.MONGO_PKG_FILE_NAME_728_TGZ;
+import static org.sonatype.nexus.plugins.helm.internal.HelmITSupport.PKG_PATH;
+import static org.sonatype.nexus.plugins.helm.internal.HelmITSupport.TGZ_EXT;
+import static org.sonatype.nexus.repository.http.HttpStatus.OK;
+import static org.sonatype.nexus.testsuite.testsupport.FormatClientSupport.status;
+
+public class CleanupTaskHelmIT
+    extends CleanupITSupport
+{
+  public static final String[] NAMES = {MONGO_PKG_FILE_NAME_404_TGZ};
+
+  public Repository repository;
+
+  @Rule
+  public RepositoryRuleHelm repos = new RepositoryRuleHelm(() -> repositoryManager);
+
+  @Configuration
+  public static Option[] configureNexus() {
+    return NexusPaxExamSupport.options(
+        NexusITSupport.configureNexusBase(),
+        nexusFeature("org.sonatype.nexus.plugins", "nexus-repository-helm")
+    );
+  }
+
+  @Before
+  public void setup() {
+    testData.addDirectory(NexusPaxExamSupport.resolveBaseFile("target/test-classes/helm"));
+    repository = repos.createHelmHosted(testName.getMethodName());
+    deployArtifacts(NAMES);
+  }
+
+  @Test
+  public void cleanupByLastBlobUpdated() throws Exception {
+    assertLastBlobUpdatedComponentsCleanedUp(repository, (long) NAMES.length,
+        () -> deployArtifacts(MONGO_PKG_FILE_NAME_728_TGZ), 1L);
+  }
+
+  @Test
+  public void cleanupByLastDownloaded() throws Exception {
+    assertLastDownloadedComponentsCleanedUp(repository, (long) NAMES.length,
+        () -> deployArtifacts(MONGO_PKG_FILE_NAME_728_TGZ), 1L);
+  }
+
+  @Test
+  public void cleanupByRegex() throws Exception {
+    assertCleanupByRegex(repository, NAMES.length, "bin.*-[4,7].*.tgz",
+        () -> deployArtifacts(MONGO_PKG_FILE_NAME_600_TGZ, MONGO_PKG_FILE_NAME_728_TGZ), 1L);
+  }
+
+  @Override
+  protected boolean componentMatchesByVersion(final Component component, final String version) {
+    return version
+        .equals(format("%s_%s%s", component.name().toLowerCase(), component.version().toLowerCase(), TGZ_EXT));
+  }
+
+  private int deployArtifacts(final String... names) {
+    try {
+      HelmClient client = new HelmClient(clientBuilder().build(),
+          clientContext(),
+          resolveUrl(nexusUrl, format("/repository/%s/", repository.getName())).toURI()
+      );
+
+      for (String name : names) {
+        assertThat(status(client.put(format("%s/%s", PKG_PATH, name), new ByteArrayEntity(getBytesFromTestData(name)))),
+            is(OK));
+      }
+
+      return names.length;
+    }
+    catch (Exception e) {
+      log.error("", e);
```

Comment:
Maybe throw `RuntimeException` to fail the test in this case? Cause return code is not checked in `setup()` for example